meilisearch-http/src/routes/indexes/documents.rs

```diff
@@ -227,26 +230,54 @@ async fn document_addition(
 
     let (uuid, mut update_file) = index_scheduler.create_update_file()?;
 
-    // TODO: this can be slow, maybe we should spawn a thread? But the payload isn't Send+Sync :weary:
-    // push the entire stream into a `Vec`.
-    // If someone sends us a never ending stream we're going to block the thread.
-    // TODO: Maybe we should write it to a file to reduce the RAM consumption
-    // and then reread it to convert it to obkv?
-    let mut buffer = Vec::new();
+    let temp_file = match NamedTempFile::new() {
+        Ok(temp_file) => temp_file,
+        Err(e) => {
+            error!("create a temporary file error: {}", e);
+            return Err(MeilisearchHttpError::Payload(ReceivePayloadErr));
+        }
+    };
+    debug!("temp file path: {:?}", temp_file.as_ref());
+    let buffer_file = match temp_file.reopen() {
+        Ok(buffer_file) => buffer_file,
+        Err(e) => {
+            error!("reopen payload temporary file error: {}", e);
+            return Err(MeilisearchHttpError::Payload(ReceivePayloadErr));
+        }
+    };
```

Comment:
I don't understand why we're doing that either?
Shouldn't a simple `as_file_mut` do the trick?

Or, if we can use a normal `TempFile` then we can delete this whole block.

Reply:
`as_file_mut` returns a reference to` file`, `tokio::fs::file` requires file. tempfile will have both read and write handles, just using temp_file is not enough, so it need `reopen` . I made some adjustments, you can review it again.